### PR TITLE
FIX Vault Query defaults to UNCONSUMED in all QueryCriteria types

### PR DESCRIPTION
**Pull request by josecoll**
Fix https://github.com/corda/corda/issues/949 by providing a default StateStatus argument to all QueryCriteria types.

### CODE REVIEW

**Comment by josecoll** [comment]
Improving code - hold fire from review!

File: core/src/main/kotlin/net/corda/core/node/services/vault/QueryCriteria.kt [hunk 1]
```diff
@@ -26,32 +26,35 @@ sealed class QueryCriteria {
     @CordaSerializable
     data class TimeCondition(val type: TimeInstantType, val predicate: ColumnPredicate<Instant>)
 
+    abstract class CommonQueryCriteria(open val status: Vault.StateStatus) : QueryCriteria() {
```
Comment (rick-r3) [inline core/src/main/kotlin/net/corda/core/node/services/vault/QueryCriteria.kt]: You always override `status` I think?  In which case just make it abstract and then don't need to pass in the constructor for no reason etc.

File: core/src/main/kotlin/net/corda/core/node/services/vault/QueryCriteria.kt [hunk 2]
```diff
@@ -26,32 +26,35 @@ sealed class QueryCriteria {
...
      */
-    data class VaultQueryCriteria @JvmOverloads constructor (
-            val status: Vault.StateStatus = Vault.StateStatus.UNCONSUMED,
-            val contractStateTypes: Set<Class<out ContractState>>? = null,
-            val stateRefs: List<StateRef>? = null,
-            val notaryName: List<X500Name>? = null,
-            val includeSoftlockedStates: Boolean = true,
-            val timeCondition: TimeCondition? = null) : QueryCriteria() {
-
+    data class VaultQueryCriteria @JvmOverloads constructor (override val status: Vault.StateStatus = Vault.StateStatus.UNCONSUMED,
```
Comment (rick-r3) [inline core/src/main/kotlin/net/corda/core/node/services/vault/QueryCriteria.kt]: Do we need any documentation updates or does this now align the behaviour with the docs?

Comment (josecoll) [inline core/src/main/kotlin/net/corda/core/node/services/vault/QueryCriteria.kt]: I'll make it clearer in the docs.

**Pull request merged**